nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/CopyAzureBlobStorage_v12.java

```diff
@@ -0,0 +1,388 @@
+
+        final boolean createContainer = context.getProperty(AzureStorageUtils.CREATE_CONTAINER).asBoolean();
+        final AzureStorageConflictResolutionStrategy conflictResolution = AzureStorageConflictResolutionStrategy.valueOf(context.getProperty(AzureStorageUtils.CONFLICT_RESOLUTION).getValue());
+
+        long startNanos = System.nanoTime();
+        try {
+            BlobServiceClient storageClient = getStorageClient(context, DESTINATION_STORAGE_CREDENTIALS_SERVICE, flowFile);
+            BlobContainerClient containerClient = storageClient.getBlobContainerClient(destinationContainerName);
+            if (createContainer && !containerClient.exists()) {
+                containerClient.create();
+            }
+
+            BlobClient blobClient = containerClient.getBlobClient(destinationBlobName);
+            Map<String, String> attributes = new HashMap<>();
+            applyStandardBlobAttributes(attributes, blobClient);
+            final boolean ignore = conflictResolution == AzureStorageConflictResolutionStrategy.IGNORE_RESOLUTION;
+
+            final BlobRequestConditions destinationRequestConditions = new BlobRequestConditions();
+
+            try {
+                if (conflictResolution != AzureStorageConflictResolutionStrategy.REPLACE_RESOLUTION) {
+                    destinationRequestConditions.setIfNoneMatch("*");
+                }
+
+                final AzureStorageCredentialsService_v12 fromCredentialsService = getCopyFromCredentialsService(context);
+                BlobServiceClient fromStorageClient = getStorageClient(context, SOURCE_STORAGE_CREDENTIALS_SERVICE, flowFile);
+                BlobContainerClient fromContainerClient = fromStorageClient.getBlobContainerClient(sourceContainerName);
+                BlobClient fromBlobClient = fromContainerClient.getBlobClient(sourceBlobName);
+
+                AzureStorageCredentialsDetails_v12 credentialsDetails = fromCredentialsService.getCredentialsDetails(flowFile.getAttributes());
+                String sourceUrl = fromBlobClient.getBlobUrl();
+                final BlobProperties fromBlobProperties = fromBlobClient.getProperties();
+                final long blobSize = fromBlobProperties.getBlobSize();
+
+                final BlobRequestConditions sourceRequestConditions = new BlobRequestConditions();
+                sourceRequestConditions.setIfMatch(fromBlobProperties.getETag());
+
+                HttpAuthorization httpAuthorization;
+                final String sasToken = (credentialsDetails.getCredentialsType() == AzureStorageCredentialsType.ACCOUNT_KEY)
+                        ? generateSas(fromContainerClient)
+                        : credentialsDetails.getSasToken();
+                if (sasToken != null) {
+                    sourceUrl += "?" + sasToken;
+                    httpAuthorization = null;
+                } else {
+                    httpAuthorization = getHttpAuthorization(credentialsDetails);
+                }
```

Comment:
Recommend reversing the logic for better readability:
```suggestion
                if (sasToken == null) {
                    httpAuthorization = getHttpAuthorization(credentialsDetails);
                } else {
                    sourceUrl += "?" + sasToken;
                    httpAuthorization = null;
                }
```